metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -35,52 +39,76 @@ case class ScalafixProvider(
       scalaVersion: ScalaVersion,
       scalaBinaryVersion: ScalaBinaryVersion,
       classpath: JList[Path]
-  ): List[l.TextEdit] = {
+  ): Future[List[l.TextEdit]] = {
     val fileInput = file.toInput
     val unsavedFile = file.toInputFromBuffers(buffers)
     if (isUnsaved(unsavedFile.text, fileInput.text)) {
       scribe.info(s"Organize imports requires saving the file first")
       languageClient.showMessage(
         MessageType.Warning,
-        s"Save ${file.toNIO.getFileName} before organizing imports"
+        s"Save ${file.toNIO.getFileName} to compile it before organizing imports"
       )
-      Nil
+      Future.successful(Nil)
     } else {
-      val scalafixConfPath = workspace.resolve(scalafixConfigPath)
-      val scalafixConf =
-        if (scalafixConfPath.isFile) Some(scalafixConfPath.toNIO)
-        else None
-      val scalafixEvaluation = for {
-        api <- getScalafix(scalaBinaryVersion)
-        urlClassLoaderWithExternalRule <- getRuleClassLoader(
-          scalaBinaryVersion,
-          api.getClass.getClassLoader
-        )
-      } yield {
-        val scalacOption =
-          if (scalaBinaryVersion == "2.13") "-Wunused:imports"
-          else "-Ywarn-unused-import"
-
-        api
-          .newArguments()
-          .withScalaVersion(scalaVersion)
-          .withClasspath(classpath)
-          .withToolClasspath(urlClassLoaderWithExternalRule)
-          .withConfig(scalafixConf.asJava)
-          .withRules(List(organizeImportRuleName).asJava)
-          .withPaths(List(file.toNIO).asJava)
-          .withSourceroot(workspace.toNIO)
-          .withScalacOptions(Collections.singletonList(scalacOption))
-          .evaluate()
-      }
+      compilations.compilationFinished(file).map { _ =>
+        val scalafixConfPath = workspace.resolve(scalafixConfigPath)
+        val scalafixConf =
+          if (scalafixConfPath.isFile) Some(scalafixConfPath.toNIO)
+          else None
+        val scalafixEvaluation = for {
+          api <- getScalafix(scalaBinaryVersion)
+          urlClassLoaderWithExternalRule <- getRuleClassLoader(
+            scalaBinaryVersion,
+            api.getClass.getClassLoader
+          )
+        } yield {
+          val scalacOption =
+            if (scalaBinaryVersion == "2.13") "-Wunused:imports"
+            else "-Ywarn-unused-import"
+
+          api
+            .newArguments()
+            .withScalaVersion(scalaVersion)
+            .withClasspath(classpath)
+            .withToolClasspath(urlClassLoaderWithExternalRule)
+            .withConfig(scalafixConf.asJava)
+            .withRules(List(organizeImportRuleName).asJava)
+            .withPaths(List(file.toNIO).asJava)
+            .withSourceroot(workspace.toNIO)
+            .withScalacOptions(Collections.singletonList(scalacOption))
+            .evaluate()
+        }
+        val edits = for {
+          results <- scalafixEvaluation
+          if results.isSuccessful()
```

Comment:
But in the other hand, it's ok since we only use scalafix for organizeimport, which doesn't report lint errors. And I will change this behavior in next version of scalafix to stick to that.

Reply:
Let me leave the note there maybe and we can followe up after the next scalafix release? I would try to fix even temporarilly if possible.